File: src/1026-Maximum-Difference-Between-Node-and-Ancestor/1026.py
class Solution:
    def maxAncestorDiff(self, root: TreeNode, mn=float('inf'), mx=0) -> int:
        return max(self.maxAncestorDiff(root.left, min(mn, root.val), max(mx, root.val)), \
            self.maxAncestorDiff(root.right, min(mn, root.val), max(mx, root.val))) \
            if root else mx - mn 

File: src/1026-Maximum-Difference-Between-Node-and-Ancestor/1026.cpp
static int x = []() {std::ios::sync_with_stdio(false); cin.tie(0); return 0; }();
class Solution 
{
public:
    int maxAncestorDiff(TreeNode* root, int mx=INT_MIN, int mn=INT_MAX) 
    {
        return root == nullptr ? mx - mn : max(maxAncestorDiff(root->left, max(mx, root->val), min(mn, root->val)),
                                              maxAncestorDiff(root->right, max(mx,root->val), min(mn, root->val)));
    }
};